samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -327,37 +303,21 @@ static void base_recv_cb(struct bt_bap_broadcast_sink *sink, const struct bt_bap
 	printk("Received BASE with %d subgroups from broadcast sink %p\n",
 	       bt_bap_base_get_subgroup_count(base), sink);
 
-#if defined(CONFIG_TARGET_BROADCAST_CHANNEL)
-	/**
-	 * Get a 32-bit bitfield of BIS indexes that cover the channel allocation values in
-	 * CONFIG_TARGET_BROADCAST_CHANNEL.
-	 */
-	printk("Target channel location: 0x%x\n", CONFIG_TARGET_BROADCAST_CHANNEL);
-	err = bt_bap_base_foreach_subgroup(base, subgroup_get_valid_bis_indexes_cb,
-					   &base_bis_index_bitfield);
-	if ((err != 0 && err != -ECANCELED) ||
-	    (err == -ECANCELED && base_bis_index_bitfield == 0)) {
-		printk("Failed to get valid BIS indexes: %d\n", err);
+	(void)memset(&base_subgroup, 0, sizeof(base_subgroup));
 
-		return;
-	}
-#else
-	err = bt_bap_base_get_bis_indexes(base, &base_bis_index_bitfield);
+	/* Get BIS index data for each subgroup */
+	err = bt_bap_base_foreach_subgroup(base, subgroup_get_valid_bis_indexes_cb, &base_subgroup);
 	if (err != 0) {
-		printk("Failed to get BIS indexes: %d\n", err);
+		printk("Failed to get valid BIS indexes: %d\n", err);
 
 		return;
 	}
 
-#endif /* CONFIG_TARGET_BROADCAST_CHANNEL */
-
-	bis_index_bitfield = base_bis_index_bitfield & bis_index_mask;
-
-	printk("bis_index_bitfield = 0x%08x\n", bis_index_bitfield);
-
 	if (broadcast_assistant_conn == NULL) {
 		/* No broadcast assistant requesting anything */
-		requested_bis_sync = BT_BAP_BIS_SYNC_NO_PREF;
+		for (uint8_t i = 0; i < CONFIG_BT_BAP_BASS_MAX_SUBGROUPS; i++) {
```

Comment:
```suggestion
		for (int i = 0; i < CONFIG_BT_BAP_BASS_MAX_SUBGROUPS; i++) {
```

Suggest to use same type for the iterator and the maximum value